### ISSUE TITLE
Detecting Drakvuf by 'remapped_gfn'

### ISSUE THREAD

**Issue by skvl** [tklengyel/drakvuf#309]
Hello.

In `libdrakvuf/vmi.c:inject_trap_pa()` new page is added inside guest physical address space like this:
```c++
        remapped_gfn->r = ++(drakvuf->max_gpfn);

        rc = xc_domain_populate_physmap_exact(drakvuf->xen->xc, drakvuf->domID, 1, 0, 0, &remapped_gfn->r);
```

Could kernel mode code scan memory after the end of virtual RAM (for example for VM with 1G of RAM scan address starting from 0x40000000) to detect this pages?

**Comment by tklengyel** [comment]
It could detect that there is a page there but reading/writing to it would return nothing. See https://github.com/tklengyel/drakvuf/blob/master/src/libdrakvuf/vmi.c#L1199

**Comment by saimoon** [comment]
Drakvuf could be detected using its shadow pages mapping to zero page:
`
        xc_altp2m_change_gfn(drakvuf->xen->xc, drakvuf->domID,
                         drakvuf->altp2m_idr, remapped_gfn->r, drakvuf->zero_page_gfn);
`
As tklengyel said in the code comments (issue and possible solution):
`
    /*
     * TODO: We will use the idr view to map all shadow pages to the zero (empty) page in case
     * something is trying to check the contents of these pages. However, since all shadow pages
     * will point to the zero page, if someone writes to one, the change will appear through the
     * other shadow pages as well, thus potentially revealing the presence of DRAKVUF. This can
     * be avoided if we cache all pages separately that have been written to and use emulate with
     * custom read data to only return the change in the page on the gfn it was written to.
     */
`
It'd be interesting to implement the proposed solution.

**Comment by tklengyel** [comment]
Certainly, feel free to do so ;)

**Comment by tklengyel** [comment]
Fixed by #412
